### PR TITLE
feat(vault): Create a Symfony Command to handle credentials migration

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to Create a Symfony Command to handle credentials migration.
This will give opportunities to user to migrate all their password macros and SNMP Community from the database to a vault.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Security/Vault/Application/UseCase/MigrateAllCredentials/CredentialMigrator.php [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                if (
+                    $credential->type === CredentialTypeEnum::TYPE_HOST
+                    || $credential->type === CredentialTypeEnum::TYPE_HOST_TEMPLATE
+                ) {
+                    $recordInformation = $this->migrateHostAndHostTemplateCredentials(
+                        $credential,
+                        $existingUuids
+                    );
+                } else {
+                    $recordInformation = $this->migrateServiceCredentials($credential, $existingUuids);
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/Vault/Application/UseCase/MigrateAllCredentials/CredentialMigrator.php:86]: rename method to handle serviceTemplate

File: centreon/src/Core/Macro/Application/Repository/ReadHostMacroRepositoryInterface.php [hunk 2]
```diff
@@ -48,4 +48,13 @@ public function findByHostIds(array $hostIds): array;
      * @return Macro[]
      */
     public function findByHostId(int $hostId): array;
+
```
Comment (jeremyjaouen) [inline centreon/src/Core/Macro/Application/Repository/ReadHostMacroRepositoryInterface.php:51]: Add in comment of the interface, that the host and host template are handled by this interface

File: centreon/src/Core/Macro/Application/Repository/ReadServiceMacroRepositoryInterface.php [hunk 3]
```diff
@@ -37,4 +37,13 @@ interface ReadServiceMacroRepositoryInterface
      * @return Macro[]
      */
     public function findByServiceIds(int ...$serviceIds): array;
+
```
Comment (jeremyjaouen) [inline centreon/src/Core/Macro/Application/Repository/ReadServiceMacroRepositoryInterface.php:40]: Add in comment of the interface, that the service and service template are handled by this interface

File: centreon/src/Core/Security/Vault/Infrastructure/Command/MigrateAllCredentials/MigrateAllCredentialsPresenter.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                        "<error>    - Type: {$this->convertTypeToString($result->type)}</>",
+                        "<error>    - Resource ID: {$result->resourceId}</>",
+                        "<error>    - Error: {$result->message}</>",
+                    ]);
+                }
+            }
+        }
+    }
+
+    private function convertTypeToString(CredentialTypeEnum $type): string {
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/Vault/Infrastructure/Command/MigrateAllCredentials/MigrateAllCredentialsPresenter.php:62]: phpdoc

**Review by adr-mo — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by a-launois — DISMISSED** [review]

File: centreon/doc/API/centreon-api.yaml [hunk 5]
```diff
@@ -8,6 +8,7 @@ info:
     + Added host_category filter to /monitoring/hostgroups endpoint
     + Added new filters for the /monitoring/hosts/categories endpoint
     + Added new filters for the /monitoring/services/categories endpoint
+    + Remove snmp_community property for the hosts and host templates listing endpoints
```
Comment (smau-centreon) [inline centreon/doc/API/centreon-api.yaml:11]:
```suggestion
    + Remove snmp_community property from the hosts and host templates listing endpoints
```

File: centreon/src/Core/Macro/Application/Repository/ReadHostMacroRepositoryInterface.php [hunk 6]
```diff
@@ -25,6 +25,9 @@
 
 use Core\Macro\Domain\Model\Macro;
 
+/**
+ * This Interface is designed to read macros for both hosts and host templates.
```
Comment (smau-centreon) [inline centreon/src/Core/Macro/Application/Repository/ReadHostMacroRepositoryInterface.php:29]:
```suggestion
 * This interface is designed to read macros for both hosts and host templates.
```

File: centreon/src/Core/Macro/Application/Repository/ReadServiceMacroRepositoryInterface.php [hunk 7]
```diff
@@ -25,6 +25,9 @@
 
 use Core\Macro\Domain\Model\Macro;
 
+/**
+ * This Interface is designed to read macros for both services and service templates.
```
Comment (smau-centreon) [inline centreon/src/Core/Macro/Application/Repository/ReadServiceMacroRepositoryInterface.php:29]:
```suggestion
 * This interface is designed to read macros for both services and service templates.
```

**Review by smau-centreon — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**